lib/util/git/auth.ts

```diff
@@ -34,28 +36,82 @@ export function getGitAuthenticatedEnvironmentVariables(
     }
   }
 
-  const gitUrlWithToken = getUrlWithToken(gitUrl, hostType, token);
+  const authenticationRules = getAuthenticationRulesWithToken(
+    originalGitUrl,
+    hostType,
+    token
+  );
 
   // create a shallow copy of the environmentVariables as base so we don't modify the input parameter object
   // add the two new config key and value to the returnEnvironmentVariables object
-  // increase the CONFIG_COUNT by one and add it to the object
-  return {
+  // increase the CONFIG_COUNT by one for each rule and add it to the object
+  const newEnvironmentVariables = {
     ...environmentVariables,
-    [`GIT_CONFIG_KEY_${gitConfigCount}`]: `url.${gitUrlWithToken}.insteadOf`,
-    [`GIT_CONFIG_VALUE_${gitConfigCount}`]: gitUrl,
-    GIT_CONFIG_COUNT: (gitConfigCount + 1).toString(),
   };
+  authenticationRules.forEach(({ url, insteadOf }) => {
+    newEnvironmentVariables[
+      `GIT_CONFIG_KEY_${gitConfigCount}`
+    ] = `url."${url}".insteadOf`;
+    newEnvironmentVariables[`GIT_CONFIG_VALUE_${gitConfigCount}`] = insteadOf;
+    gitConfigCount++;
+  });
+  newEnvironmentVariables['GIT_CONFIG_COUNT'] = gitConfigCount.toString();
+
+  return newEnvironmentVariables;
 }
 
-function getUrlWithToken(
-  gitUrl: string,
+function getAuthenticationRulesWithToken(
+  url: string,
   hostType: string,
   authToken: string
-): string {
+): AuthenticationRule[] {
   let token = authToken;
   if (hostType === PlatformId.Gitlab) {
-    token = `gitlab-ci-token:${token}`;
+    token = `gitlab-ci-token:${authToken}`;
   }
+  return getAuthenticationRules(url, token);
+}
+
+/**
+ * Generates the authentication rules for later git usage for the given host
+ * @link https://coolaj86.com/articles/vanilla-devops-git-credentials-cheatsheet/
```

Comment:
As the link suggests, we should use `GIT_ASKPASS` for the tokens, so they never got written to disk.

This can be done in a future PR. We should open a new issue for it.

Reply:
https://github.com/renovatebot/renovate/issues/13673